### PR TITLE
[Marcus Choo] iP

### PR DESCRIPTION
**Pull request by marcusc55**
# Duke Chatbot 

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke is a chatbot written in **Java**! Duke helps you to remember and manage your tasks.

Duke
1. is easy to use
2. supports multiple types of tasks
   1. To-dos
   2. Deadlines
   3. Events

and Duke can help you to:
* Add tasks
* View tasks
* Remove tasks
* Mark tasks as done
* Show your tasks
* Search for tasks

You can download Duke from [here](https://github.com/marcusc55/ip/releases/)!

Features:
- [x] Managing tasks
- [x] Managing deadlines and events with dates
- [ ] GUI support (Coming soon!)

***
If you are a Java programmer, you can use it to practice Java too! Here's the `Launcher` method:
```java
public class Launcher { 
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

### CODE REVIEW

File: src/main/java/duke/TaskList.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import java.util.stream.Collectors;
+
+import duke.task.Task;
+
+/**
+ * This class represents Duke's list of tasks.
+ */
+public class TaskList {
+    private static final String STORAGE_PATH = "data/duke.txt";
+    private List<Task> list = new ArrayList<>();
```
Comment (sw-boi) [inline src/main/java/duke/TaskList.java]: Maybe rename this List<Task> as "tasks"? To represent a list of Task objects.

**Review by sw-boi — COMMENTED** [review]
LGTM. I really like how clean your code is. 😄 I aspire to code like this.

File: src/main/java/duke/Main.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package duke;
+
+public class Main {
+    /**
+     * Main class to start the Duke chatbot.
```
Comment (Tianqi-Zhu) [inline src/main/java/duke/Main.java]: Should be an empty line between description and parameters.

File: src/main/java/duke/command/Command.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import duke.DukeException;
+import duke.Parser;
+
+/**
+ * This class encapsulates a user's command.
+ */
+public abstract class Command {
+    public static final Command[] COMMANDS = new Command[] { new DeadlineCommand(), new DeleteCommand(),
+        new DoneCommand(), new EndCommand(), new EventCommand(), new FindCommand(), new ListCommand(),
+        new TodoCommand() };
```
Comment (Tianqi-Zhu) [inline src/main/java/duke/command/Command.java]: Indentation for wrapped lines should be 8 spaces.

File: src/main/java/duke/TaskList.java (hunk 1 above)
Comment (marcusc55) [reply src/main/java/duke/TaskList.java]: Thanks for spotting this, will rename this.

File: src/main/java/duke/Main.java (hunk 2 above)
Comment (marcusc55) [reply src/main/java/duke/Main.java]: Thanks for spotting this, fixed in a later commit!

File: src/main/java/duke/command/Command.java (hunk 3 above)
Comment (marcusc55) [reply src/main/java/duke/command/Command.java]: Thanks for spotting this, fixed in a later commit!

**Pull request closed without merging**